app/portainer/views/account/create-user-access-token/create-user-access-token.controller.js

```diff
@@ -0,0 +1,39 @@
+export default class CreateUserAccessTokenController {
+  /* @ngInject */
+  constructor($async, $state, $analytics, Authentication, UserService, Notifications) {
+    this.$async = $async;
+    this.$state = $state;
+    this.$analytics = $analytics;
+    this.Authentication = Authentication;
+    this.UserService = UserService;
+    this.Notifications = Notifications;
+
+    this.onSubmit = this.onSubmit.bind(this);
+    this.onSuccess = this.onSuccess.bind(this);
+    this.onError = this.onError.bind(this);
+  }
+
+  async onSubmit(description) {
+    const accessToken = await this.UserService.createAccessToken(this.state.userId, description);
+    // Dispatch analytics event upon success accessToken generation
+    this.$analytics.eventTrack('portainer-account-access-token-create', { category: 'portainer' });
```

Comment:
why not just do it in `onSuccess`?

Reply:
`onSuccess` is called when you click the done button; you may just generate the access token and not click done (which will prevent this analytics from being captured)

Reply:
ok, so onSuccess sounds like on the success of creating the token. maybe rename to `onDoneClick`? to be explicit.

Although if you use the router in react you don't need onSuccess or onDone

Reply:
using suggested react router now 👍